### PR TITLE
Fix bugs in InterpolatePointCloudToRegularGrid and MapPointCloudToRegularGrid, Update QMMeltPoolImporter

### PR DESCRIPTION
**Pull request by imikejackson**
Also allows QMMeltPoolImporter to use more than a single HDF5 file.

Also some code cleanups were also applied into this PR.

Related PRs
https://github.com/BlueQuartzSoftware/DREAM3D/pull/966
https://github.com/BlueQuartzSoftware/SIMPLView/pull/71
https://github.com/BlueQuartzSoftware/SIMPL/pull/415

### CODE REVIEW

**Comment by imikejackson** [comment]
https://github.com/BlueQuartzSoftware/DREAM3DCI/pull/53
https://github.com/BlueQuartzSoftware/DREAM3DCI/pull/52

File: DREAM3DReviewFilters/ImportQMMeltpoolH5File.cpp [hunk 1]
```diff
@@ -125,19 +127,19 @@ std::string make_list_string(const Container& items)
 
 struct ImportQMMeltpoolH5File::Cache
 {
-  QDateTime lastModified;
-  QString filePath;
+  // QDateTime lastModified;
```
Comment (JDuffeyBQ) [inline DREAM3DReviewFilters/ImportQMMeltpoolH5File.cpp]:
```suggestion
```

File: DREAM3DReviewFilters/ImportQMMeltpoolH5File.cpp [hunk 2]
```diff
@@ -125,19 +127,19 @@ std::string make_list_string(const Container& items)
...
   std::vector<int64_t> missingIndices;
-
   std::vector<int64_t> layerThicknesses;
   std::vector<int64_t> numElements;
 
   void flush()
   {
     filePath = "";
-    lastModified = QDateTime();
+    // lastModified = QDateTime();
```
Comment (JDuffeyBQ) [inline DREAM3DReviewFilters/ImportQMMeltpoolH5File.cpp]:
```suggestion
```

File: DREAM3DReviewFilters/ImportQMMeltpoolH5File.cpp [hunk 3]
```diff
@@ -165,16 +166,87 @@ void ImportQMMeltpoolH5File::initialize()
   setCancel(false);
 }
 
+// -----------------------------------------------------------------------------
+std::pair<int32_t, std::string> checkFile(const std::string& filePath)
```
Comment (JDuffeyBQ) [inline DREAM3DReviewFilters/ImportQMMeltpoolH5File.cpp]: This function should be moved to the anonymous namespace.

Comment (imikejackson) [reply DREAM3DReviewFilters/ImportQMMeltpoolH5File.cpp]: Done

**Pull request merged**